### PR TITLE
Add command for numeric reply

### PR DESCRIPTION
**Pull request by toqoz**
cf. http://www.haun.org/kent/lib/rfc1459-irc-ja.html#c2.4
## Sample

``` ruby
require 'zircon'

client = Zircon.new(
  server: '127.0.01',
  port: 6667,
  channel: '#channel',
  username: 'dummy',
)

client.on_numericreply do |msg|
  puts "(#{msg.type}) #{msg.body}"
end

client.run!
```
## Output

```
(001) Welcome to the debian Internet Relay Chat Network dummy
(002) Your host is hybrid7.debian.local[127.0.0.1/6667], running version hybrid-7.2.2.dfsg.2-6
(003) This server was created Feb  8 2010 at 21:14:28
(004) hybrid7.debian.local
(005) CALLERID
(005) CHANLIMIT=#&:15
(251) There are 0 users and 3 invisible on 1 servers

...
```

### CODE REVIEW

File: lib/zircon/callback.rb [hunk 1]
```diff
@@ -7,7 +7,9 @@ module Callback
     private
 
     def method_missing(method_name, *args, &block)
-      if method_name =~ /(list|on|trigger)_([a-z0-9]+)/
+      if method_name =~ /(list|on|trigger)_([0-9]+)/
+        send($1, 'numericreply', *args, &block)
+      elsif method_name =~ /(list|on|trigger)_([a-z0-9]+)/
```
Comment (r7kamura) [inline lib/zircon/callback.rb]: In this case, the case-when statement is useful

``` ruby
case method_name
when /(list|on|trigger)_([0-9]+)/
  ...
when /(list|on|trigger)_([a-z0-9]+)/
  ...
else
 ...
end
```

Comment (toqoz) [inline lib/zircon/callback.rb]: Thanks! I have been rewrite by using it.

**Comment by r7kamura** [comment]
Thanks, I've bumped up the version from v0.0.4 to v0.0.5 :+1:

**Comment by toqoz** [comment]
I will use its version soon in ircmad :+1:

**Pull request merged**